### PR TITLE
Implement AgentDispatch service

### PR DESCRIPTION
**Pull request by biglittlebigben**
This allows listing, adding and deleting agent dispatches on an existing room. Requests go to a new AgentDispatchService, which sends them over RPC to the rtc.Room via the RoomManager. The rtc.Room then does agent job management using RPCs to the agent service.

### CODE REVIEW

File: pkg/rtc/room.go [hunk 1]
```diff
@@ -146,6 +151,88 @@ type ParticipantOptions struct {
...
+
+type agentJob struct {
+	*livekit.Job
+	lock sync.Mutex
+	done chan struct{}
+}
+
+// This provides utilities attached the agent dispatch to ensure that all pending jobs are created
+// before terminating jobs attached to an agent dispatch. This avoids a race that could cause some pending jobs
+// to not be terninated when a dispatch is deleted.
```
Comment (paulwe) [inline pkg/rtc/room.go:168]: terminated

File: pkg/rtc/room.go [hunk 2]
```diff
@@ -146,6 +151,88 @@ type ParticipantOptions struct {
...
+func (j *agentJob) participantLeft() {
+	j.lock.Lock()
+	if j.done != nil {
+		close(j.done)
+		j.done = nil
+	}
+	j.lock.Unlock()
+}
+
+func (j *agentJob) waitForParticipantLeaving(mu *sync.RWMutex) error {
```
Comment (paulwe) [inline pkg/rtc/room.go:217]: what is mu for?

File: pkg/rtc/room.go [hunk 3]
```diff
@@ -199,7 +288,7 @@ func NewRoom(
 
 	r.createAgentDispatchesFromRoomAgent()
```
Comment (paulwe) [inline pkg/rtc/room.go:289]: can this be async?

File: pkg/service/agent_dispatch_service.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	}
+}
+
+func (ag *AgentDispatchService) CreateDispatch(ctx context.Context, req *livekit.CreateAgentDispatchRequest) (*livekit.AgentDispatch, error) {
+	err := EnsureAdminPermission(ctx, livekit.RoomName(req.Room))
+	if err != nil {
+		return nil, twirpAuthError(err)
+	}
+
+	return ag.agentDispatchClient.CreateDispatch(ctx, rpc.RoomTopic(req.Room), req)
```
Comment (paulwe) [inline pkg/service/agent_dispatch_service.go:40]: need an `rpc.TopicFormatter` in this service (see `RoomService`) to support cloud topics

File: pkg/service/roommanager.go [hunk 5]
```diff
@@ -807,6 +817,58 @@ func (r *RoomManager) UpdateRoomMetadata(ctx context.Context, req *livekit.Updat
 	return room.ToProto(), nil
 }
 
+func (r *RoomManager) ListDispatch(ctx context.Context, req *livekit.ListAgentDispatchRequest) (*livekit.ListAgentDispatchResponse, error) {
+	room := r.GetRoom(ctx, livekit.RoomName(req.Room))
+	if room == nil {
+		return nil, ErrRoomNotFound
+	}
+
+	room.Logger.Debugw("room ListDispatch", "dispatchID", req.DispatchId)
```
Comment (paulwe) [inline pkg/service/roommanager.go:826]: server rpc logging should already give us these

File: pkg/agent/worker.go [hunk 6]
```diff
@@ -374,12 +409,21 @@ func (w *Worker) handleAvailability(res *livekit.AvailabilityResponse) {
 }
 
 func (w *Worker) handleJobUpdate(update *livekit.UpdateJobStatus) {
+	err := w.updateJobStatus(update)
+
+	if err != nil {
+		w.logger.Infow("received job update for unknown job", "jobId", update.JobId)
```
Comment (paulwe) [inline pkg/agent/worker.go:415]: `jobID`

**Review by paulwe — APPROVED** [review]

File: pkg/rtc/room.go (hunk 2 above)
Comment (biglittlebigben) [reply pkg/rtc/room.go:217]: A leftover from a refactor... I removed it.

File: pkg/rtc/room.go (hunk 3 above)
Comment (biglittlebigben) [reply pkg/rtc/room.go:289]: We probably want createAgentDispatchesFromRoomAgent to have run by the time NewRoom returns, as its output is needed by the agents lifecycle RPCs. It does very little though (the only potentially blocking call is to storage). 

launchRoomAgents could be run in its own goroutine, but it already instantiate goroutines internally do do actual work

**Pull request merged**